src/Service/GlobalAuthenticationService.php

```diff
@@ -57,10 +65,23 @@ class GlobalAuthenticationService extends ID3BaseService
 
     /**
      * @var array Optional extras for the SOAP client
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.PropertyTypeHint.MissingTraversableTypeHintSpecification
      */
     private array $soapOptions = [];
 
-    public function __construct(GlobalAuthenticationGateway $gateway, array $soapOptions = [])
+    /**
+     * When constructing this class, you can either first construct a GlobalAuthenticationGateway and pass it in, or you
+     * can leave this blank to use the default GlobalAuthenticationGateway. In the former case, you should set the API
+     * username/password and SOAP options directly on the gateway when you create it. Otherwise, you should set that on
+     * this service class before calling ->verifyIdentity().
+     *
+     * @param GlobalAuthenticationGateway|null $gateway Either the gateway class to use, or null to have the service
+     * instantiate this gateway for you.
+     * @param array $soapOptions If you don't pass in a gateway and you want to configure options for the SoapClient,
+     * you can pass these in here, or call ->setSoapOptions() after constructing the object.
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingTraversableTypeHintSpecification
+     */
+    public function __construct(?GlobalAuthenticationGateway $gateway, array $soapOptions = [])
```

Comment:
A default `null` value should be set for the `$gateway` variable if the constructor is intended to be used without giving any parameter.

Reply:
Good spotting, my mistake for missing that. Ta!